dex/wait/queue.go

```diff
@@ -238,17 +238,11 @@ func (q *TaperingTickerQueue) Run(ctx context.Context) {
 			w := waiters[0]
 			waiters = waiters[1:]
 			wg.Add(1)
-			go runWaiter(w)
+			go runWaiter(w) // note: async, so the next one could run its TryFunc first!
 
 		case w := <-q.queueWaiter:
-			// A little optimization if this waiter would fire immediately, but
-			// it works to append regardless.
-			if time.Until(w.nextTick) <= 0 {
-				wg.Add(1)
-				go runWaiter(w)
-				continue
-			}
-
```

Comment:
I actually want to keep this. Is there any way we can adjust the test instead? What about doing this in `TestTaperingQueue`?

```Go
if numTrys > numTryAgains*2 {
```

Reply:
I'd like to keep it too.  Your test seem to enforce an implicit design goal of maintaining order, which I was trying to keep, but I think that can (needs to?) change too.